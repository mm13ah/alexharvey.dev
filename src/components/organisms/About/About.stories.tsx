import React from 'react';
import { StoryFn, Meta } from '@storybook/react';

import { About } from './About';

export default {
  title: 'Organisms/About',
  component: About,
  parameters: {
    layout: 'fullscreen',
  },
} as Meta<typeof About>;

const Template: StoryFn<typeof About> = () => <About />;

export const Primary = Template.bind({});
Primary.storyName = 'About';
