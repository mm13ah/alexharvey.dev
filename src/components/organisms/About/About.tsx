import Image from 'next/image';
import React from 'react';
import { Background } from '../../atoms';

import AboutPic from '../../../../public/about.png';

export const About = () => {
  return (
    <Background
      id="about"
      className={`bg-white grid grid-cols-1 lg:grid-cols-2 p-10 dark:bg-gradient-to-r dark:from-gray-900
      dark:via-stone-700 dark:to-gray-900 dark:border-b-2 dark:border-solid dark:border-gray-700`}
    >
      <div className="flex flex-row items-center justify-center w-full h-full show-on-scroll">
        <Image
          alt="me"
          src={AboutPic}
          width={0}
          height={0}
          className="w-auto h-full p-10"
        />
      </div>
      <div className="lg:p-10 show-on-scroll">
        <h2 className="text-neutral-900 font-medium tracking-wider text-4xl mb-2 dark:text-white">
          About
        </h2>
        <p className="text-neutral-900 text-xl tracking-wide leading-relaxed font-light dark:text-white">
          Hi, I&apos;m Alex, an agile software developer currently working at{' '}
          <a
            href="https://www.contic.co.uk/"
            target="_blank"
            rel="noreferrer"
            className="underline text-blue-500 dark:text-cyan-500"
          >
            Contic
          </a>
          . I love working with JavaScript and all the brilliant open source
          technologies that come with it. In my day-to-day life, I work
          primarily with TypeScript, React, GraphQL, NodeJS and PostgreSQL. I
          also have a solid foundation in writing semantic HTML and applying
          responsive CSS.
        </p>
        <br />
        <p className="text-neutral-900 text-xl tracking-wide leading-relaxed font-light dark:text-white">
          Outside of work, I enjoy playing gigs with my band,{' '}
          <a
            href="https://open.spotify.com/artist/1HDp77de4Mrv8OnnEduNJu"
            target="_blank"
            rel="noreferrer"
            className="underline text-blue-500 dark:text-cyan-500"
          >
            Pretty Terry
          </a>
          , DJing, producing music and speaking bad French avec l&apos;aide de
          ma petite amie.
        </p>
        <br />
        <p className="text-neutral-900 text-xl tracking-wide leading-relaxed font-light dark:text-white">
          If you&apos;d like to know more about my techical experience, you can{' '}
          <a
            href="#experience"
            className="underline text-blue-500 dark:text-cyan-500"
          >
            check it out below
          </a>
          .
        </p>
      </div>
    </Background>
  );
};
