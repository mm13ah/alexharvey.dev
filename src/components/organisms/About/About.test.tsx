import React from 'react';
import { render } from '@testing-library/react';
import renderer from 'react-test-renderer';
import { About } from './About';

describe('About', () => {
  it('should render', () => {
    render(<About />);
  });
  it('should match a snapshot', () => {
    const tree = renderer.create(<About />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
