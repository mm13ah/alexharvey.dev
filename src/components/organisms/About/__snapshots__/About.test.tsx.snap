// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`About should match a snapshot 1`] = `
<section
  className="w-screen min-h-screen flex bg-white grid grid-cols-1 lg:grid-cols-2 p-10 dark:bg-gradient-to-r dark:from-gray-900
      dark:via-stone-700 dark:to-gray-900 dark:border-b-2 dark:border-solid dark:border-gray-700"
  id="about"
>
  <div
    className="flex flex-row items-center justify-center w-full h-full show-on-scroll"
  >
    <img
      alt="me"
      className="w-auto h-full p-10"
      data-nimg="1"
      decoding="async"
      height={40}
      loading="lazy"
      onError={[Function]}
      onLoad={[Function]}
      src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
      srcSet="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
      style={
        Object {
          "color": "transparent",
        }
      }
      width={40}
    />
  </div>
  <div
    className="lg:p-10 show-on-scroll"
  >
    <h2
      className="text-neutral-900 font-medium tracking-wider text-4xl mb-2 dark:text-white"
    >
      About
    </h2>
    <p
      className="text-neutral-900 text-xl tracking-wide leading-relaxed font-light dark:text-white"
    >
      Hi, I'm Alex, an agile software developer currently working at
       
      <a
        className="underline text-blue-500 dark:text-cyan-500"
        href="https://www.contic.co.uk/"
        rel="noreferrer"
        target="_blank"
      >
        Contic
      </a>
      . I love working with JavaScript and all the brilliant open source technologies that come with it. In my day-to-day life, I work primarily with TypeScript, React, GraphQL, NodeJS and PostgreSQL. I also have a solid foundation in writing semantic HTML and applying responsive CSS.
    </p>
    <br />
    <p
      className="text-neutral-900 text-xl tracking-wide leading-relaxed font-light dark:text-white"
    >
      Outside of work, I enjoy playing gigs with my band,
       
      <a
        className="underline text-blue-500 dark:text-cyan-500"
        href="https://open.spotify.com/artist/1HDp77de4Mrv8OnnEduNJu"
        rel="noreferrer"
        target="_blank"
      >
        Pretty Terry
      </a>
      , DJing, producing music and speaking bad French avec l'aide de ma petite amie.
    </p>
    <br />
    <p
      className="text-neutral-900 text-xl tracking-wide leading-relaxed font-light dark:text-white"
    >
      If you'd like to know more about my techical experience, you can
       
      <a
        className="underline text-blue-500 dark:text-cyan-500"
        href="#experience"
      >
        check it out below
      </a>
      .
    </p>
  </div>
</section>
`;
